test/utils/InYearUtilSpec.scala

```diff
@@ -40,32 +40,32 @@ class InYearUtilSpec extends UnitTest {
 
   "InYearAction.inYear" when {
 
-    "return true when taxYear is 2022 and current date is 5th April 2022 and time is just before midnight" in {
-      val currentDate: LocalDateTime = LocalDateTime.of(year2022, month4, day5, hour23, minute59)
-      inYearAction.inYear(year2022, currentDate) shouldBe true
+    s"return true when taxYear is $yearCurrent and current date is 5th April $yearCurrent and time is just before midnight" in {
+      val currentDate: LocalDateTime = LocalDateTime.of(yearCurrent, month4, day5, hour23, minute59)
+      inYearAction.inYear(yearCurrent, currentDate) shouldBe true
     }
 
-    "return false when taxYear is 2022 and current date is 6th April 2022 at midnight" in {
-      val currentDate: LocalDateTime = newDate(year2022, month4, day6)
-      inYearAction.inYear(year2022, currentDate) shouldBe false
+    s"return false when taxYear is $yearCurrent and current date is 6th April $yearCurrent at midnight" in {
+      val currentDate: LocalDateTime = newDate(yearCurrent, month4, day6)
+      inYearAction.inYear(yearCurrent, currentDate) shouldBe false
     }
 
-    "return false when taxYear is 2022 and current date is 6th April 2022 at one minute past midnight" in {
-      val currentDate: LocalDateTime = LocalDateTime.of(year2022, month4, day6, 0, 1)
-      inYearAction.inYear(year2022, currentDate) shouldBe false
+    s"return false when taxYear is $yearCurrent and current date is 6th April $yearCurrent at one minute past midnight" in {
+      val currentDate: LocalDateTime = LocalDateTime.of(yearCurrent, month4, day6, 0, 1)
+      inYearAction.inYear(yearCurrent, currentDate) shouldBe false
     }
 
-    "return false when taxYear is 2022 and current date is 6th April 2022 at one hour past midnight" in {
-      val currentDate: LocalDateTime = LocalDateTime.of(year2022, month4, day6, 1, 1)
-      inYearAction.inYear(year2022, currentDate) shouldBe false
+    s"return false when taxYear is $yearCurrent and current date is 6th April $yearCurrent at one hour past midnight" in {
+      val currentDate: LocalDateTime = LocalDateTime.of(yearCurrent, month4, day6, 1, 1)
+      inYearAction.inYear(yearCurrent, currentDate) shouldBe false
     }
   }
 
   "InYearAction.notInYear" when {
 
     def block: Future[Result] = { Future.successful(Results.Ok("Success Response")) }
 
-    "current date is before April 6th 2021 which means the tax year is 2020-21" should {
+    s"current date is before April 6th 2021 which means the tax year is 2020-21" should {
```

Comment:
the tax years in the test names can be dynamic too

Reply:
I left these one's as the tests below weren't dynamic ones